airflow/pipe_events_dag.py

```diff
@@ -1,58 +1,39 @@
+    'encounters': imp.load_source('pipe_events_encounters', pp.join(pp.dirname(__file__), 'pipe_events_encounters_dag.py')),
+    'anchorages': imp.load_source('pipe_events_anchorages', pp.join(pp.dirname(__file__), 'pipe_events_anchorages_dag.py')),
+    'fishing': imp.load_source('pipe_events_fishing', pp.join(pp.dirname(__file__), 'pipe_events_fishing_dag.py')),
+}
 
 
-class GapEventsDagFactory(DagFactory):
+class PipelineDagFactory(DagFactory):
     def __init__(self, pipeline=PIPELINE, **kwargs):
-        super(GapEventsDagFactory, self).__init__(pipeline=pipeline, **kwargs)
-
-    def source_date_range(self):
-        # shift the date range one day back
-        if self.schedule_interval == '@daily':
-            return '{{ yesterday_ds }}', '{{ yesterday_ds }}'
-        elif self.schedule_interval == '@monthly':
-            start_date = '{{ (execution_date.replace(day=1) + macros.dateutil.relativedelta.relativedelta(days=-1)).strftime("%Y-%m-%d") }}'
-            end_date = '{{ (execution_date.replace(day=1) + macros.dateutil.relativedelta.relativedelta(months=1, days=-2)).strftime("%Y-%m-%d") }}'
-            return start_date, end_date
-        else:
-            raise ValueError('Unsupported schedule interval {}'.format(self.schedule_interval))
+        super(PipelineDagFactory, self).__init__(pipeline=pipeline, **kwargs)
 
     def build(self, dag_id):
-        config = self.config
-        config['source_table'] = config['position_messages']
-        config['date_range'] = ','.join(self.source_date_range())
-
         with DAG(dag_id, schedule_interval=self.schedule_interval, default_args=self.default_args) as dag:
-            source_sensors = self.source_table_sensors(dag)
-
-            publish_events = BashOperator(
-                task_id='publish_events',
-                pool='bigquery',
-                bash_command='{docker_run} {docker_image} gap_events '
-                             '{date_range} '
-                             '{project_id}:{source_dataset}.{position_messages} '
-                             '{project_id}:{events_dataset}.{gap_events_table} '
-                             '{gap_events_min_pos_count} '
-                             '{gap_events_min_dist}'.format(**config)
-            )
-
-            for sensor in source_sensors:
-                dag >> sensor >> publish_events
-
-            return dag
-
-
-gaps_events_daily_dag = GapEventsDagFactory().build('gap_events_daily')
-gaps_events_monthly_dag = GapEventsDagFactory(schedule_interval='@monthly').build('gap_events_monthly')
+            for (subdag_name, dag_factory) in subpipelines:
+                subdag_id = '{}.{}'.format(PIPELINE, subdag_name)
+                subdag_factory = dag_factory.PipelineDagFactory(
+                    pipeline=subdag_id, schedule_interval=self.schedule_interval, extra_config=self.config)
+
+                return SubDagOperator(
+                    subdag=subdag_factory.build(dag_id=subdag_id),
```

Comment:
@andres-arana I think it is worth mentioning something that I experienced with SubDagOperator. It's that the Subdag will be shown as `running` until the actual `SubDag` completes, no matter if the `SubDag` is in `retry` state. The behavior I was waiting for was that the `SubDagOperator` task to be displayed as `retry` instead of `running`. I didn't find any workaround to this yet. I don't think it will cause any problem to you, but just for you to know that you may experience that.

Reply:
@andres-arana I've been looking into this. I don't know if you want to implement what I'm going to say next. Since we are not specifying the number of retries in the `Subdag` bash operator task, the `SubdagOperator` task will fail when the default max_retry criteria is met. My recomendation is to set a fix number in the BashOperator, maybe `3` for starters. Once it met that number, the `BashOperator` will fail and the `SubDagOperator` will be marked as up-for-retry.

Reply:
Oh, I see. Ok, I'll do that then.